### ISSUE TITLE
examples/window/shaders.metal is very wrong

### ISSUE THREAD

**Issue by OneSadCookie** [gfx-rs/metal-rs#75]
the binary metallib checked into the repository works, but this .metal file cannot be compiled to produce that metallib.

This is closer:
```metal
// code: language=c++
#include <metal_stdlib>

using namespace metal;

typedef struct {
    float x, y, r, g, b;
} vertex_t;

static_assert(alignof(vertex_t) == 4, "vertex_t must match a rust array of f32");
static_assert(sizeof(vertex_t) == 4 * 5, "vertex_t must match a rust array of f32");

struct ColorInOut {
    float4 position [[position]];
    float4 color;
};

// vertex shader function
vertex ColorInOut triangle_vertex(device vertex_t* vertex_array [[ buffer(0) ]],
                                   uint32_t vid [[ vertex_id ]])
{
    ColorInOut out;
    
    auto device const &v = vertex_array[vid];
    out.position = float4(v.x, v.y, 0.0, 1.0);
    out.color = float4(v.r, v.g, v.b, 1.0);
    
    return out;
}

// fragment shader function
fragment float4 triangle_fragment(ColorInOut in [[stage_in]])
{
    return in.color;
}
```

I suspect the example should include a build.rs to compile the metallib from source to avoid this problem recurring.

**Comment by OneSadCookie** [comment]
(notable problems fixed: "vertex" and "fragment" keywords were flipped on the functions, layout of the vertex array didn't match the rust code)

**Comment by kvark** [comment]
oops, nicely spotted! We should be doing it in `build.rs` all the time and not include the binary in the repo

**Comment by kvark** [comment]
Fixed by #111
